testing/tests/DevExpress.ui.widgets/treeViewParts/lazyRendering.js

```diff
@@ -22,6 +26,25 @@ QUnit.test("Only root nodes should be rendered by default", function(assert) {
     assert.equal(items.length, 2);
 });
 
+QUnit.test("Expanding nodes should work with special charactes in id", function(assert) {
+    let testCases = ['!/#$%&\'()"+./:;<=>?@[]^`{|}~\\,', '____2______.jpg', 'E:\\test\\[gsdfgfd]  |  \'[some__file]', '!@#$%^&*()_+'];
+    each(testCases, (index, testId) => {
+        var $treeView = initTree({
+            dataSource: [
+                { id: testId, text: "item1", selected: false, expanded: false },
+                { id: 'aaaa', parentId: testId, text: "item2", selected: false, expanded: false }
+            ],
+            dataStructure: "plain",
+            height: 500
+        });
+        assert.equal(false, $treeView.find('[aria-level="2"]').is(':visible'));
+
+        $treeView.find('[aria-level="1"]').find('.dx-treeview-toggle-item-visibility').trigger('dxclick.dxTreeView');
+        assert.equal(true, $treeView.find('[aria-level="2"]').is(':visible'));
+        $treeView.dxTreeView('instance').dispose();
+    });
+});
+
```

Comment:
Please, check the nodes selections and filtering of the TreeView in the tests with the specific keys

Reply:
Great idea.
Fixed